src/Core/src/Handlers/Button/ButtonHandler.iOS.cs

```diff
@@ -202,9 +211,56 @@ public override void SetImageSource(UIImage? platformImage)
 				if (Handler?.PlatformView is not UIButton button)
 					return;
 
-				platformImage = platformImage?.ImageWithRenderingMode(UIImageRenderingMode.AlwaysOriginal);
+				nfloat buttonSize = 0;
+
+#pragma warning disable CA1422 // Validate platform compatibility
+				var contentEdgeInsets = button.ContentEdgeInsets;
+#pragma warning restore CA1422 // Validate platform compatibility
+
+				if (button.Bounds != CGRect.Empty)
+				{
+					if (button.Bounds.Height > button.Bounds.Width)
+						buttonSize = button.Bounds.Width - (contentEdgeInsets.Left + contentEdgeInsets.Right);
+					else
+						buttonSize = button.Bounds.Height - (contentEdgeInsets.Top + contentEdgeInsets.Bottom);
+				}
+
+				try
+				{
+					if (buttonSize != 0)
+						platformImage = ResizeImageSource(platformImage, buttonSize, buttonSize);
+
+					platformImage = platformImage?.ImageWithRenderingMode(UIImageRenderingMode.AlwaysOriginal);
+
+					button.SetImage(platformImage, UIControlState.Normal);
+				}
+				catch (Exception)
+				{
+					Handler.MauiContext?.CreateLogger<ButtonHandler>()?.LogWarning("Can not load Button ImageSource");
+				}
+			}
+
+			static UIImage? ResizeImageSource(UIImage? sourceImage, nfloat maxWidth, nfloat maxHeight)
```

Comment:
There is already a ScaleImage extension method that does this. It also has overloads to take in a max sze and fit the image into that. And, there is even an official Scale method on UIIMage itself - not quite sure why we rewrote that. Seems to be just in Xamarin not in official UIKit... Weird.

Reply:
I have not done the replacement with the existing extension method because it also created a new UIImage by rescaling. I have applied changes to keep the same image by changing the scale.